api/agent.go

```diff
@@ -1053,7 +1053,14 @@ func (a *Agent) ForceLeavePrune(node string) error {
 // ForceLeaveOpts is used to have the agent eject a failed node or remove it
 // completely from the list of members.
 func (a *Agent) ForceLeaveOpts(node string, opts ForceLeaveOpts) error {
```

Comment:
@huikang : Is there anything to update in the API docs for this endpoint? It seems like this PR is meant to correct intended behavior (to be consistent with the docs), rather than to change behavior / deviate from the docs.

Reply:
I've added a test
I didn't want to remove `ForceLeaveOpts` to avoid breaking anyone currently using it.
The Consul API doesn't change, and it already supports reading and utilizing an ACL token. It's just the client side that didn't have this option.

Let me know if something else is needed.

Reply:
@jkirschner-hashicorp , you are right. No change is needed for API doc.

Reply:
We could mark `ForceLeaveOpts` deprecated.

```go
// ForceLeaveOpts is used to have the agent eject a failed node or remove it
// completely from the list of members.
// 
// DEPRECATED - Use ForceLeaveOptions instead.
func (a *Agent) ForceLeaveOpts(node string, opts ForceLeaveOpts) error
```

Reply:
marked